src/compiler/scanner.rs

```diff
@@ -63,7 +63,27 @@ pub fn skip_trivia(text: &str, pos: isize) -> isize {
         return pos;
     }
 
-    pos
+    let mut pos = pos as usize;
```

Comment:
I believe these are safe `usize`/`isize` conversions but would probably be better to use the `.try_from()` versions or whatever would panic if it can't actually correctly convert between the two